### PR TITLE
Update to Swift 5

### PR DESCRIPTION
**Pull request by mattt**
This PR updates the package manifest to `swift-tools-version:5.3` makes the following changes for Swift 5:

- Removes dependency on [FisherYates](https://github.com/apple/example-package-fisheryates), now that we have a built-in [`shuffle` method](https://developer.apple.com/documentation/swift/array/2994753-shuffle) on `Array`.
  - The original purpose of these examples was to have a tree that showed a mix of direct and transitive dependencies. I think we can arrive at a similar result by adding a dependency on `swift-argument-parser` to `dealer`.
- Removing unnecessary `XCTestManifests` (tests now require `--enable-test-discovery`)
- Miscellaneous stylistic refinements

### CODE REVIEW

**Comment by tomerd** [comment]
sorry @mattt could you rebase?

**Comment by mattt** [comment]
@tomerd Sure thing! Done.

File: Tests/LinuxMain.swift [hunk 1]
```diff
@@ -1,16 +1 @@
...
- See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
- */
-
-import DeckOfPlayingCardsTests
-import XCTest
-
-var tests = [XCTestCaseEntry]()
-tests += DeckOfPlayingCardsTests.allTests()
-XCTMain(tests)
+fatalError("Run with `swift test --enable-test-discovery`")
```
Comment (tomerd) [inline Tests/LinuxMain.swift]: should we just remove this file? its not needed

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
cc @neonichu @abertelrud

**Comment by tomerd** [comment]
thanks for this @mattt  <3

**Review by neonichu — APPROVED** [review]

**Comment by neonichu** [comment]
Should https://github.com/apple/example-package-fisheryates be archived after this? Seems like it doesn't have much of a point on its own.

**Comment by mattt** [comment]
> Should https://github.com/apple/example-package-fisheryates be archived after this? Seems like it doesn't have much of a point on its own.

Yes, I think that would be the right thing to do.

I have a draft PR for the `dealer` package that updates to Swift 5.4 and adds a dependency on `swift-argument-parser` (which is not only an exemplary library for any Swift CLI, but a nice addition to the package graph for purposes or demonstrating dependency resolution). See https://github.com/apple/example-package-dealer/pull/7

**Comment by tomerd** [comment]
> Should https://github.com/apple/example-package-fisheryates be archived after this? Seems like it doesn't have much of a point on its own.

done

**Pull request merged**